core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -593,6 +593,23 @@ assert equalType(
     return substitutions;
   }
 
+  /**
+   * Equivalence checking for row types, but except for the field names.
+   */
+  private boolean rowTypesAreEquivalent(
+      MutableRel rel0, MutableRel rel1, Litmus litmus) {
+    // Validation checking for row type, but except for the field names.
+    assert rel0.rowType.getFieldCount() == rel1.rowType.getFieldCount()
```

Comment:
This comment seems redundant, because there is already a java doc comment for the method.